### PR TITLE
Various small fixes/tweaks

### PR DESCRIPTION
**Pull request by classicrocker883**
### Description
Most file changes remove `HAS_STATUS_MESSAGE` swap w/`HAS_DISPLAY`
Also, 2nd most are changes to whitespace, and comments.

- remove whitespace/fix wording, tweak spacing  
- ~~add hyphen (-) to "Gcode" -- (follow up to "📝 GCode => G-Code")~~ i believe done in another **PR**  
- ~~remove unused `INLINE_USART_IRQ`~~  done in another **PR**  
- optimize code - add `MMS_SCALED` to `feedrate_mm_s` + simplified the PID Plot code in **proui/dwin.cpp** so it's easier to read
- add a Default-No Probe/No Mesh test to **buildroot/tests/STM32F103RE_creality**
- Removed `COLOR_AQUA` because it is unused, and also the same as `COLOR_CYAN`
- add `HAS_TRAMMING_WIZARD`
- ~~`set_status_and_level(..., level=0)` from #26181~~ done in another **PR** 
<br>

- - this should be looked at if this change is needed vs change in other **PR**(see *Note*)
- - same thing change for **M48.cpp**
<sup>~~without `level=0`, we can just add `, 0` like so `set_status_and_level(...code.., 0)`??~~</sup>
- ~~`extrude_length` => `purge_length` to match others~~ done in another **PR** 
- `dtostrf` => `p/w_float_t` -- ~~**M114.cpp**~~|**M48.cpp**

Update
---

follow up to "[BUG] DWIN ProUI: error: 'PROBE_OFFSET_ZMIN' was not declared in this scope #26505"  
in _ProUI_, created `_OFFSET_ZMIN/MAX` like in is in _CrealityUI_, made it so it = `PROBE_OFFSET_ZMIN/MAX` if `HAS_BED_PROBE`/defined, otherwise = `+/-20`

- optimize #define/#include in JyersUI and CrealityUI as well
- use `PROBE_SELECTED` to enabled `PROBE_OFFSET_ZMIN|MAX` (because --see below--`)
- simplified Print Progress (`_printtime`, `_remain_time`, `_percent_done`) in ProUI
<sup>may need to adjust layout position on the UI</sup>  
- - made use of `duration_t` -- `toDigital`, and `pcttostrpctrj`
- - benefits: less bytes, also adds Days to print time elapsed/remaining
- update **bedlevel_tools.cpp**, utilize unused functions
- add function to `zeroPoint` (reset single mesh x/y coordinate)
- update trammingwizard:  
  shows correct screen while inuse - rather than user being able to still navigate menu  
  exits the wizard - displays within tolerance - if values are 0.00 as fail-safe, otherwise gives false reading.
  show's Lower/Raise in green/red.
- move `dwinIconShow` to respected header
- add to ESDiag: filament sensor "PRESENT" (green) / "Runout Detected" (red)
- - rename `ES_REPORT` => `ES_REPORTS` to not be confused with previously defined in **endstops.cpp**
- remove/replace `dwinPopupConfirm` w/ `dwinPopupContinue`
- typo in **module/probe.cpp** `Probe::run_z_probe()`. "-float" to "- float"
- swap `const char *` with `PGM_P` (which is defined as `const char *`)
- Tweak `CASELIGHT_USES_BRIGHTNESS` in **ProUI**, separate function that sets it into "apply" and "live"

- fix tramming wizard, so now:
>When x = 0 and y = 0, p will be calculated as p = 0 + 2 * 0 = 0.  
>When x = 0 and y = 1, p will be calculated as p = 1 + 2 * 0 = 1.  
>When x = 1 and y = 0, p will be calculated as p = 0 + 2 * 1 = 2.  
>When x = 1 and y = 1, p will be calculated as p = 1 + 2 * 1 = 3.

>[!NOTE]
>~~*I'm not so sure about this one*~~
> - ~~in **pinsDebug_list.h** I added Maple support which have their SDIO defined as `BOARD_SDIO_*`~~
```diff
 #if PIN_EXISTS(SDIO_D0)
   REPORT_NAME_DIGITAL(__LINE__, SDIO_D0_PIN)
+#elif defined(BOARD_SDIO_D0)
+  REPORT_NAME_DIGITAL(__LINE__, BOARD_SDIO_D0)
 #endif
```

<br>

in **Marlin\src\inc\Conditionals_post.h**  
you will see
```y
#if !HAS_LEVELING || ANY(MESH_BED_LEVELING, AUTO_BED_LEVELING_UBL)
  #undef PROBE_MANUALLY
#endif
#if ANY(HAS_BED_PROBE, PROBE_MANUALLY, MESH_BED_LEVELING)
  #define PROBE_SELECTED 1
#endif
```

---
removed `HAS_ONESTEP_LEVELING`, replaced with `HAS_AUTOLEVEL` because  
(previously...)

```y
#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT, AUTO_BED_LEVELING_UBL)
  #define HAS_ONESTEP_LEVELING 1
#endif
```
this code is the same as `HAS_AUTOLEVEL`, as you can see here:
```y
#if ANY(HAS_ABL_NOT_UBL, AUTO_BED_LEVELING_UBL)
  #define HAS_ABL_OR_UBL 1
  #if DISABLED(PROBE_MANUALLY)
    #define HAS_AUTOLEVEL 1
  #endif
#endif
```
<br>

### Related Issues
if anyone wishes to take another look, these are what I suggest:  
see if **buildroot/tests/STM32F103RE_creality** needs anything  

and also...

in JyersUI/**dwin.cpp**  
`MAX_Z_OFFSET` seemed out of place without `MIN_Z_OFFSET`
basically these values can be used even without `HAS_Z_OFFSET_ITEM` (`BABYSTEPPING`), it would cause an error being undefined for some Configurations

```diff
-#if HAS_ZOFFSET_ITEM
-  #define MAX_Z_OFFSET 9.99
-  #if HAS_BED_PROBE
-    #define MIN_Z_OFFSET -9.99
-  #else
-    #define MIN_Z_OFFSET -1
-  #endif
-#endif

+#if HAS_ZOFFSET_ITEM && HAS_BED_PROBE
+  #define MAX_Z_OFFSET  20
+  #define MIN_Z_OFFSET -20
+#else
+  #define MAX_Z_OFFSET  3
+  #define MIN_Z_OFFSET -3
+#endif
```

### CODE REVIEW

File: docs/Cutter.md [hunk 1]
```diff
@@ -51,10 +51,10 @@ The following flow charts depict the flow control logic for spindle and laser op
...
-    ┌─────┐  │  │M3 I      │  │G0,G1,G2,G4│  │Planner    │  │Planner     │  │Planner fan │  │Planner     │  │Stepper    │
-    │GCode│  │  │Continuous│  │M3 receive │  │sets block │  │sync power ?│  │sync power ?│  │trap power ?│  │uses block │
-    │Send ├──┼──┤Inline    ├──┤power from ├──┤power using├──┤process M3  ├──┤process fan ├──┤adjusts for ├──┤values to  │
-    └─────┘  │  │          │  │S-Value    │  │Gx S-Value │  │power inline│  │power inline│  │accel/decel │  │apply power│
+   ┌──────┐  │  │M3 I      │  │G0,G1,G2,G4│  │Planner    │  │Planner     │  │Planner fan │  │Planner     │  │Stepper    │
+   │G-Code│  │  │Continuous│  │M3 receive │  │sets block │  │sync power ?│  │sync power ?│  │trap power ?│  │uses block │
+   │Send  ├──┼──┤Inline    ├──┤power from ├──┤power using├──┤process M3  ├──┤process fan ├──┤adjusts for ├──┤values to  │
+   └──────┘  │  │          │  │S-Value    │  │Gx S-Value │  │power inline│  │power inline│  │accel/decel │  │apply power│
              │  └──────────┘  └───────────┘  └───────────┘  └────────────┘  └────────────┘  └────────────┘  └───────────┘
```
Comment (thisiskeithb) [inline docs/Cutter.md]: This is broken now:

image

There's also a "gcode" in: https://github.com/MarlinFirmware/Marlin/blob/483b8dcc05c2f0ef4c83d8ec7f4f0065ec9d240f/docs/Cutter.md?plain=1#L7

Fixed in https://github.com/MarlinFirmware/Marlin/pull/26502/commits/87dbb8e20747d8c1df404371f6c70f0dd109ba2b

**Comment by thisiskeithb** [comment]
> made it so to #undef PROBE_OFFSET_ZMIN/MAX when DWIN_LCD_PROUI is disabled. because this value is used when editing Z Offset. even without a probe or leveling, and babystepping is enabled

Sounds like these Ender-3 V2 UIs need to decouple from requiring/assuming a probe is installed instead of this workaround.

**Comment by classicrocker883** [comment]
> Sounds like these Ender-3 V2 UIs need to decouple from requiring/assuming a probe is installed instead of this workaround.

it uses that probe offset value regardless of one or not, using it for changing the Z-offset.

how about using a `TERN` in the function? 
```diff
- setPFloatOnClick(PROBE_OFFSET_ZMIN, PROBE_OFFSET_ZMAX, 2, applyZOffset, liveZOffset);
+ setPFloatOnClick(TERN(PROBE_OFFSET_ZMIN, 20), TERN(PROBE_OFFSET_ZMAX, 20), 2, applyZOffset, liveZOffset);
```
or completely omit this #undef since the other `PROBE_OFFSET_` arent included?

or we could add somewhere

```y
#ifndef PROBE_OFFSET_ZMIN
  #define PROBE_OFFSET_ZMIN 20
#endif
#ifndef PROBE_OFFSET_ZMAX
  #define PROBE_OFFSET_ZMAX 20
#endif
```

File: Marlin/src/lcd/e3v2/proui/dwin.h [hunk 2]
```diff
@@ -234,7 +234,7 @@ void doCoolDown();
 #if ENABLED(NOZZLE_PARK_FEATURE)
   void parkHead();
 #endif
-#if HAS_ONESTEP_LEVELING
+#if HAS_AUTOLEVEL
   void trammingwizard();
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin.h]: Note that tramming is not leveling. And "one step" leveling is any `G29` procedure that can be completed without human intervention. So the tramming wizard should not be associated with `HAS_AUTOLEVEL` or `HAS_ONESTEP_LEVELING`.

It may help to bring @mriscoc into the conversation about changes to ProUI since some of these ProUI flags may be modified to use existing configuration options that apply to other UIs.

File: Marlin/src/inc/Conditionals_post.h [hunk 3]
```diff
@@ -2946,7 +2946,11 @@
 /**
  * Bed Probe dependencies
  */
-#if ANY(MESH_BED_LEVELING, HAS_BED_PROBE)
+#if ANY(BABYSTEPPING, PROBE_SELECTED)
+  #define HAS_ZOFFSET_ITEM 1
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_post.h:2952]: Different UIs are going to have different criteria for whether or not they display a "Z Offset" item, so this flag should be left with each individual UI that currently uses it.

File: Marlin/src/inc/Conditionals_post.h [hunk 4]
```diff
@@ -2946,7 +2946,11 @@
...
+#endif
+
+#if HAS_ZOFFSET_ITEM
   #ifndef PROBE_OFFSET_ZMIN
     #define PROBE_OFFSET_ZMIN -20
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_post.h:2957]: These value range settings should only pertain to editing of the Probe Z Offset. Note that a UI may include both a baby-stepping item that has no effect on the probe offset, and a Probe Z Offset adjustment that doesn't baby-step. Non-probe-offset-adjusting baby-stepping is not limited to any range.

File: Marlin/src/lcd/e3v2/creality/dwin.cpp [hunk 5]
```diff
@@ -1139,7 +1131,7 @@ void popupWindowHome(const bool parking/*=false*/) {
   }
 }
 
-#if HAS_ONESTEP_LEVELING
+#if HAS_AUTOLEVEL
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/creality/dwin.cpp]: This includes UBL but unfortunately UBL is rarely a one step leveling process. It usually requires an auto leveling step followed by a manual leveling step. So some additional checking must be done to ensure that UBL is actually one-step (i.e., it can reach all mesh points with the probe) or additional UI must be developed to guide users through the following steps.

File: Marlin/src/inc/Conditionals_post.h (hunk 3 above)
Comment (classicrocker883) [reply Marlin/src/inc/Conditionals_post.h:2952]: that may be true, however the only UIs which have `HAS_ZOFFSET_ITEM` are exactly those three. that I am able to find.

File: Marlin/src/lcd/e3v2/proui/dwin.h (hunk 2 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.h]: this is the "trammingwizard", it used the bed probe to check each corner.  
if one wanted to tram without a having a bed probe, the menu still gives that option. the wizard is just the auto version by checking level like it would for a mesh.

Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.h]: Screenshot 2023-12-08 040838

File: Marlin/src/inc/Conditionals_post.h (hunk 3 above)
Comment (thinkyhead) [reply Marlin/src/inc/Conditionals_post.h:2952]: Yep, and all those UIs are externally and independently developed, no guarantee they'll remain in sync. So these flags should be left with their respective UIs.

File: Marlin/src/lcd/e3v2/proui/dwin.h (hunk 2 above)
Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/dwin.h]: The "tramming wizard" may involve the probe, but it is a feature independent of leveling, and it should be possible to disable it even if a probe exists and leveling is enabled. These things are separate in the configurations, and I have been encouraging the authors of UI code to respect these settings in our discussions. We should continue to encourage and develop in the direction of properly applying configuration settings rather than perpetuating the oversights of UI authors.

Comment (mriscoc) [reply Marlin/src/lcd/e3v2/proui/dwin.h]: > It may help to bring @mriscoc into the conversation about changes to ProUI since some of these ProUI flags may be modified to use existing configuration options that apply to other UIs.

I currently use:
```cpp
#if ENABLED(LCD_BED_TRAMMING)
  void drawTrammingMenu();
  #if HAS_TRAMMING_WIZARD
    void runTrammingWizard();
  #endif
#endif
```
That allows me to disable the wizard independently to save program memory.

File: Marlin/src/inc/Conditionals_post.h (hunk 4 above)
Comment (classicrocker883) [reply Marlin/src/inc/Conditionals_post.h:2957]: the main concern is `DWIN_LCD_PROUI` uses `PROBE_OFFSET_ZMIN/MAX` whether it `HAS_MESH`, a probe --auto/mesh bed leveling, or none of the above. because it uses that offset value to adjust the Z-Offset (nozzle to bed relation).

I mean if this **PR** change isn't good, we could always just redefine it in **dwin_defines.h** into something else like  
```y
#ifndef PROBE_Z_OFFSET_ZMIN
  #define Z_OFFSET_ZMIN   20
#else
  #define Z_OFFSET_ZMIN  PROBE_Z_OFFSET_ZMIN
#endif
```
and use the new value in for the max/min of the function

Comment (thisiskeithb) [reply Marlin/src/inc/Conditionals_post.h:2957]: [As I mentioned above](https://github.com/MarlinFirmware/Marlin/pull/26502#issuecomment-1846055677), these UIs need to decouple from requiring/assuming a probe is installed so we don't end up with these kind of bug reports when someone doesn't have specific hardware or enable a specific feature:

- https://github.com/MarlinFirmware/Marlin/issues/26505

File: Marlin/src/lcd/e3v2/creality/dwin.cpp (hunk 5 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/creality/dwin.cpp]: so then it makes sense to be `HAS_AUTOLEVEL`?  
as far as I understand it, UBL doesnt *have* to be followed by manual leveling. the auto level option usually probes all the points, and any points not able to reach, it fills in the _unprobed_ values based on the surrounding _probed_ points, and in my experience with great accuracy.

for Bilinear or UBL, its usually the default, and one and done kind of thing. and if they wanted to check the points or do it manually they have that option as well.

**Comment by classicrocker883** [comment]
I wanna say this **PR** is ready to go. please take another look through

**Comment by classicrocker883** [comment]
ok I want to say this is ready. at least for another review.

File: Marlin/src/gcode/lcd/M0_M1.cpp [hunk 6]
```diff
@@ -73,9 +73,9 @@ void GcodeSuite::M0_M1() {
       ExtUI::onUserConfirmRequired(GET_TEXT_F(MSG_USERWAIT));
   #elif ENABLED(DWIN_LCD_PROUI)
     if (parser.string_arg)
-      dwinPopupConfirm(ICON_BLTouch, parser.string_arg, GET_TEXT_F(MSG_USERWAIT));
+      dwinPopupContinue(ICON_BLTouch, F(parser.string_arg), GET_TEXT_F(MSG_USERWAIT));
```
Comment (thinkyhead) [inline Marlin/src/gcode/lcd/M0_M1.cpp]: C strings, `PROGMEM` strings, and `F` strings all need to be treated differently. If `dwinPopupContinue` wants to do something with a C string then there must a separate `dwinPopupContinue` method that takes a C string. If we wanted a method that accepts a `PROGMEM` string (`PGM_P`) it must be named `dwinPopupContinue_P` because `PROGMEM` strings cannot be distinguished from regular C strings. `PROGMEM` strings can be wrapped in `FPSTR()` to be passed to a method taking an `F` string. And `F` strings can be wrapped in `FTOP()` to be passed to a method taking a `PROGMEM` string.

File: Marlin/src/lcd/e3v2/creality/dwin.cpp (hunk 5 above)
Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/creality/dwin.cpp]: UBL only fills in the un-probed values when commanded to do so by a call to `G29 P3`. I don't think this old Creality screen implementation does that.

File: Marlin/src/lcd/e3v2/proui/dwin_popup.h [hunk 7]
```diff
@@ -69,11 +69,3 @@ void dwinShowPopup(const uint8_t icon, T amsg1=nullptr, U amsg2=nullptr, uint8_t
   dwinDrawPopup(icon, amsg1, amsg2, button);
   dwinUpdateLCD();
 }
-
-template<typename T, typename U>
-void dwinPopupConfirm(const uint8_t icon, T amsg1, U amsg2) {
-  hmiSaveProcessID(ID_WaitResponse);
-  dwinDrawPopup(icon, amsg1, amsg2, BTN_Confirm);  // Button Confirm
-  dwinUpdateLCD();
-}
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin_popup.h:79]: Templated functions like this can be used to do double duty for C strings and F strings (assuming `dwinDrawPopup` is also set up to do both C and F strings in a non-templated way (or does checking with `__typeof__` within the templated function).

File: Marlin/src/pins/pinsDebug_list.h [hunk 8]
```diff
@@ -2300,21 +2300,36 @@
 //
 #if PIN_EXISTS(SDIO_D0)
   REPORT_NAME_DIGITAL(__LINE__, SDIO_D0_PIN)
+#elif defined(BOARD_SDIO_D0)
+  REPORT_NAME_DIGITAL(__LINE__, BOARD_SDIO_D0)
```
Comment (thinkyhead) [inline Marlin/src/pins/pinsDebug_list.h]: Only pins that are actually assigned to a Marlin function need to be reported here. We don't need to report pins defined with a `BOARD_` prefix, as they are merely intended as defaults if not otherwise defined.

File: Marlin/src/lcd/e3v2/proui/endstop_diag.cpp [hunk 9]
```diff
@@ -57,11 +57,12 @@ void draw_es_label(FSTR_P const flabel=nullptr) {
...
 
-void draw_es_state(const bool is_hit) {
+void draw_es_state(const bool is_hit, const bool is_fil) {
   const uint8_t LM = 130;
   DWINUI::cursor.x = LM;
-  dwinDrawRectangle(1, hmiData.colorPopupBg, LM, DWINUI::cursor.y, LM + 100, DWINUI::cursor.y + 20);
-  is_hit ? DWINUI::drawString(RGB(31,31,16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16,63,16), F(STR_ENDSTOP_OPEN));
+  dwinDrawRectangle(1, hmiData.colorPopupBg, LM, DWINUI::cursor.y, LM + 120, DWINUI::cursor.y + 20);
+  is_fil ? (is_hit ? DWINUI::drawString(RGB(16, 63, 16), F("PRESENT")) : DWINUI::drawString(RGB(31, 31, 16), F("Runout Detected"))) :
+  (is_hit ? DWINUI::drawString(RGB(31, 31, 16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16, 63, 16), F(STR_ENDSTOP_OPEN)));
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/endstop_diag.cpp:66]: If these are for display on the LCD to common users they should use translatable strings with `GET_TEXT_F` rather than inline strings or `STR_*` strings that are intended for serial output.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp [hunk 10]
```diff
@@ -68,10 +69,6 @@
   #include "../../../feature/host_actions.h"
 #endif
 
-#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT)
-  #define HAS_ONESTEP_LEVELING 1
-#endif
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: We should keep this define. If you've modified the LCD code to call `G29 P1\nG29 P3` for UBL then UBL can be considered "one step" but the ABL options with `PROBE_MANUALLY` are definitely not one-step leveling procedures. `PROBE_MANUALLY` requires user interaction with the encoder to measure each point and proceed to the next point on click.

Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: I do not see anywhere in ProUI that `HAS_ONESTEP_LEVELING` is being utilized. its currently defined like this in two seperate files. even the current repo at mriscoc's ProUI doesnt have that and it has no use for `PROBE_MANUALLY`. when UBL or ABL is used, its automatically understood you have that option...to use it as a "one step" or not. just like Mesh Bed Leveling.  
which is also called Manual Mesh Bed Leveling right? since its manually probing the bed without an actual probe.

what is confusing is the `#define PROBE_MANUALLY` option in configuration.h:
>The "Manual Probe" provides a means to do "Auto" Bed Leveling without a probe.  


it sounds like if you have `MESH_BED_LEVELING`, you would want to also enable this. yet in **inc/Conditionals_LCD.h**:
```y
#if !HAS_LEVELING || ANY(MESH_BED_LEVELING, AUTO_BED_LEVELING_UBL)
  #undef PROBE_MANUALLY
```

but besides that, a user having ABL or UBL should always have the option to "Probe manually" regardless.

that is how the menu is currently, except for maybe JyersUI. that is a bit confusing to have UBL, but also being able to not having any bed probe defined?

in summary, I think this should be left specifically for `MESH_BED_LEVELING`; if you dont have a probe, that should be your only option for mesh leveling.  
and in regards to Auto bed leveling w/probe, it should always give you that "probe manually" option anyway in the menu/gcode. in otherwords, if it `HAS_LEVELING` the "probing manually" is always an option. 

I'm just saying this is how it is set up with ProUI right now
- Probe?
UBL, or ABL
- No Probe?
Mesh Bed Leveling, or None

I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

It seems that things aren't what they used to be, or meant to be anymore. or maybe I have it all wrong. like was `AUTO_BED_LEVELING_UBL` supposed to be used without a bed probe like it can be in JyersUI? if so, why not just use `MESH_BED_LEVELING` instead?

on a side note, even Mriscoc is phasing out using Bilinear auto bed leveling in favor of UBL. I know their code is different, but on the surface its the same thing. *you probe, and you get a mesh.*  
plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

File: Marlin/src/lcd/e3v2/proui/endstop_diag.cpp (hunk 9 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/endstop_diag.cpp:66]: other than making up new ones, I'm looking through what is available, what about `MSG_YES` for hit, and `MSG_NO` for open?

as for "PRESENT" => `MSG_FILAMENT` and "Runout Detected" => `MSG_RUNOUT_SENSOR`, or it could be `MSG_FILAMENT_CHANGE_HEADER_LOAD` ("LOAD FILAMENT")

File: Marlin/src/lcd/e3v2/proui/dwin.cpp (hunk 10 above)
Comment (mriscoc) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: > plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

Currently the paradigm is different, due to the addition of AML 2.0 support, now all mesh parameters are saved in the UBL slot, so you can have slots with different mesh sizes and mesh densities. You can know how it is working now and the new way of use UBL in the Professional firmware here: https://github.com/mriscoc/Ender3V2S1/wiki/Mesh-Leveling-Menu#ubl

Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: I picked this line at random to comment on the define, but the general idea is to keep using this define in code that currently uses it, wherever that is.

File: Marlin/src/lcd/e3v2/proui/endstop_diag.cpp (hunk 9 above)
Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/endstop_diag.cpp:66]: Making up new ones is the correct way to go for strings not yet defined. Prefer pre-existing ones where possible. I went ahead and added a commit to internationalize various strings. If you find any more that need translation as you go along, please add them.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp (hunk 10 above)
Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: > I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

Yes, that option is specifically for a machine that has no probe. You can also use `PROBE_MANUALLY` with any ABL option (other than UBL) for a machine that has no probe. In both cases, you send a `G29` command to get each point, and keep doing this in a loop until the procedure is completed.

See `menu_bed_leveling.cpp` within the `ANY(PROBE_MANUALLY, MESH_BED_LEVELING)` block to see how this is implemented from the UI.

Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: > due to the addition of AML 2.0 support

Good that you have a wiki to cover the topic. My aim with ABL Bilinear was to make a leveling system that can be done all in one go based on a mesh that only covers the area of the bed that the probe can reach. It actually is "adaptive" in that you can set the boundaries of the mesh that you want to probe, but it always probes the same number of points.

We have a PR pending to make all leveling systems runtime-configurable in both the boundaries and the number of points in X and Y, but of course it's slow going because every LCD implementation wants to do things its own way. I cannot stress enough how important it is to work with the main Marlin project to standardize these things in the core and not to keep "going it alone." So anything you can do towards that effort would be much appreciated. I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

File: Marlin/src/pins/pinsDebug_list.h (hunk 8 above)
Comment (thinkyhead) [reply Marlin/src/pins/pinsDebug_list.h]: Currently the SDIO pins are defined by the HAL and then in some instances pins files define some `BOARD_SDIO_*` pins (without a `_PIN` suffix). These break the pattern that we had previously established, so it would be good to fix these up. The way it should work is:
- Define `BOARD_SDIO_*_PIN` pins in any pins files that want to override the HAL.
- Define `SDIO_*_PIN` as `BOARD_SDIO_*_PIN` when such a define exists,
- Otherwise, define `SDIO_*_PIN` based on the MCU.

I see that some of these are defined by `variants/*/board.h` while `pins_AQUILA_101.h` is the only pins file that does so. It's not entirely clear where in the build process `variants/*/board.h` is included, but I presume that it's ahead of all Marlin sources since it's part of the platform support. So, the common approach should work as described above for all the `SDIO_*_PIN` defines.

Comment (thinkyhead) [reply Marlin/src/pins/pinsDebug_list.h]: That said, caveats like this should not hold up this PR. Any items in this sweeping PR that are still under discussion can be moved to a separate PR as necessary.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp (hunk 10 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:75]: > I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

this quote 😂  
oh good, I'm not alone

File: Marlin/src/pins/pinsDebug_list.h (hunk 8 above)
Comment (classicrocker883) [reply Marlin/src/pins/pinsDebug_list.h]: I thought I had since reverted this. so I'll go through this once over to see if there is anything left to do. otherwise if you can point anything out, or list it what needs to be changed lmk

**Comment by thinkyhead** [comment]
Thanks for adding more languages. Take care to correct any strings that are over 20 characters, as these must be shortened to 20 characters or fewer. If a longer string is preferred for wider screens, put the wider strings into the LanguageWide namespace.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp [hunk 11]
```diff
@@ -2052,14 +2040,14 @@ void gotoConfirmToPrint() {
...
+    safe_delay(500);
     dwinUpdateLCD();
     DONE_BUZZ(settings.save());
   }
 
   void readEEPROM() {
-    const bool success = settings.load();
     dwinRedrawScreen();
-    DONE_BUZZ(success);
+    DONE_BUZZ(settings.load());
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp:2081]: `settings.load` should come ahead of directly redrawing the screen, because otherwise any screen that displays or uses settings will be displaying / using old settings.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp [hunk 12]
```diff
@@ -2069,7 +2057,10 @@ void gotoConfirmToPrint() {
   }
 
   #if HAS_MESH
-    void saveMesh() { TERN(AUTO_BED_LEVELING_UBL, ublMeshSave(), writeEEPROM()); }
+    void saveMesh() {
+      TERN_(MESH_BED_LEVELING, manualMeshSave();)
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp]: Keep semicolons at the end of lines so that linters won't bark at us. If there is some situation where the presence of a semicolon would be bad (e.g., inside of a `switch` block) then use `OPTCODE` instead of `TERN_` and the semicolon will be added automagically.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp [hunk 13]
```diff
@@ -2555,7 +2559,7 @@ void setFlow() { setPIntOnClick(FLOW_EDIT_MIN, FLOW_EDIT_MAX, []{ planner.refres
 
   void manualMeshSave() {
     LCD_MESSAGE(MSG_UBL_STORAGE_MESH_MENU);
-    queue.inject(F("M211S1\nM500"));
+    queue.inject(F("M211S1"));
```
Comment (thinkyhead) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp:2592]: This doesn't pertain to UBL. See line 2061.
```cpp
TERN_(MESH_BED_LEVELING, manualMeshSave());
```

Comment (thinkyhead) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:2592]: Also, `M211S1` enables software endstops. I'm not sure how that is pertinent here.

**Review by classicrocker883 — COMMENTED** [review]
> Thanks for adding more languages. Take care to correct any strings that are over 20 characters, as these must be shortened to 20 characters or fewer. If a longer string is preferred for wider screens, put the wider strings into the LanguageWide namespace. 

I thought I would do at least a couple, but seeing it is a bit tedious for even one; going back and forth with google translate, and another, trying to get the words right... and since I only know one real language, I figure I'll leave the rest to someone else, or maybe if I get bored enough.

File: Marlin/src/lcd/e3v2/proui/endstop_diag.cpp (hunk 9 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/endstop_diag.cpp:66]: im going to leave the `STR_*` for now because a. I can't find a suitable `MSG_` replacement, and b. im sure "TRIGGERED" in big red letters is universally understood 😆.

File: Marlin/src/lcd/e3v2/proui/dwin.cpp (hunk 13 above)
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:2592]: well, `manualMeshStart` (line 2537) disables them inorder for it to do its bed leveling. 

>This doesn't pertain to UBL. See line 2061.

sorry, I don't follow...

**Comment by thinkyhead** [comment]
It is not possible to review all these changes and accept them all in one PR. You should cherry-pick parts of this large PR into smaller pieces with changes limited to specific areas of the code, or only addressing specific things, and submit those separately as smaller PRs.
